File: code-samples/ai/code-snippets/speaker-diarization.php
<?php
$NGROK_ADDRESS = "NGROK-TUNNEL-ADDRESS";
$WEBHOOK_URL = $NGROK_ADDRESS . "/webhook";
$CONTENT_URI = "PUBLICLY-ACCESSIBLE-CONTENT-URI";

speakers_detection();

/*
* Detect speakers from a conversation
*/
function speakers_detection() {
  global $platform, $WEBHOOK_URL, $CONTENT_URI;
    try {

      $bodyParams = array (
          'contentUri' =>  $CONTENT_URI,
          'encoding' => "Mpeg",
          'languageCode' =>  "en-US",
          'source' => "RingCentral",
          'audioType' =>  "CallCenter"
      );
      $endpoint = "/ai/audio/v1/async/speaker-diarize?webhook=" . urlencode($WEBHOOK_URL);
      $resp = $platform->post($endpoint, $bodyParams);
      $jsonObj = $resp->json();
      if ($resp->response()->getStatusCode() == 202) {
        print_r ("Job ID: " . $jsonObj->jobId . PHP_EOL);
        print_r("Ready to receive response at: " . $WEBHOOK_URL . PHP_EOL);
      }
    }catch (\RingCentral\SDK\Http\ApiException $e) {
      print_r ('Unable to call speaker diarization API. ' . $e->getMessage() . PHP_EOL);
    }
}
?>

File: code-samples/ai/code-snippets/speaker-diarization.py
from ringcentral import SDK
import os,sys,urllib.parse,json

NGROK_ADDRESS = "NGROK-TUNNEL-ADDRESS"
WEBHOOK_URL = NGROK_ADDRESS + "/webhook";
CONTENT_URI = "PUBLICLY-ACCESSIBLE-CONTENT-URI"

#
# Detect speakers from a conversation
#
def speakers_detection():
    try:
        bodyParams = {
        'contentUri': CONTENT_URI,
        'encoding': "Mpeg",
        'languageCode': "en-US",
        'source': "RingCentral",
        'audioType': "CallCenter"
        }
        endpoint = f'/ai/audio/v1/async/speaker-diarize?webhook={urllib.parse.quote(WEBHOOK_URL)}'
        resp = platform.post(endpoint, bodyParams)
        jsonObj = resp.json()
        if resp.response().status_code == 202:
            print(f'Job ID: {resp.json().jobId}')
            print(f'Ready to receive response at: {WEBHOOK_URL}')
    except Exception as e:
        print ("Unable to call speaker diarization API. " + str(e))


# Authenticate a user using a personal JWT token
def login():
  try:
      platform.login( jwt= "PRODUCTION-JWT" )
      speakers_detection()
  except Exception as e:
      print ("Unable to authenticate to platform. Check credentials. " + str(e))
